### PR TITLE
Updating `cpppar_quf_table_sum()` to work with cpp11

### PR DESCRIPTION
**Pull request by thisisnic**
This PR is for comparison purposes; I would recommend *not* merging it as-is as there as some further changes to be made.
I'd recommend:

* not merging the tests at all.  They're excessive, and I have left them in for illustration purposes only (see separate discussion doc)
* creating a wrapper function for the R version of `cpppar_quf_table_sum()` which calls `as.integer()` on the inputs, so the user can supply integers or doubles as they like

### CODE REVIEW

File: src/06_01_quf_helpers.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "06_quf.h"
+#include "06_0_quf.h"
```
Comment (thisisnic) [inline src/06_01_quf_helpers.cpp:1]: I had to rename this here so it loaded that file first, as I think when I was iterating through this it had some headers I needed.

File: src/06_03_quf_tables.cpp [hunk 2]
```diff
@@ -368,33 +369,33 @@ void quf_table_sum_single(void *px_in, string &x_type, int n, int q, int *x_quf,
     res.push_back({"quf"_nm = res_x_quf_all});
 
     // x: Unik
-    writable::list res_tmp(Q);
-    res.push_back({"items"_nm = x_unik_all});
+    // items
+    writable::list res_tmp;
+    for(int i = 0; i < Q; i++){
+        res_tmp.push_back(writable::doubles(x_unik_all[i]));
+    }
```
Comment (thisisnic) [inline src/06_03_quf_tables.cpp:376]: The key thing with these loops is that we don't need to instatiate `res_tmp`'s size as it's fine to just create it empty and use `push_back()` to add things to it.  If we give it a size and then use `push_back()` we're adding extra items on top of the instantiated ones.

We also need to call `writable::doubles()` on the item we want to add in.  I found it useful to look at the `cpp11` package's unit tests to try to understand what syntax was needed here.

**Comment by pachadotdev** [comment]
this is excellent!
I still can't fit a model, but there's huge progress here

so far I see
```
cpp11::cpp_register()
devtools::install()
fixest::feols(mpg ~ wt, data = mtcars)
Error: r_vector
``` 

I added some of the files in this PR to my branch. The good point is that it builds now!

**Comment by pachadotdev** [comment]
I'll close this PR, because I added most of it by copy and paste. Thanks a lot !!!

**Pull request closed without merging**